react/DeliverySettings.tsx

```diff
@@ -246,25 +246,30 @@ const DeliverySettings: FC = () => {
   }
 
   const handleSubmit = async (showToast: any) => {
-
+    setIsLoading(true);
     if (freeDeliveryThreshold !== 0) {
       handlePromotionSubmit()
     }
 
     await updateShippingRate({variables: {
       ...shippingRate
-    }})
+    }}).catch(err=>{
+      console.error(err)
+    })
 
     await updateOnboarding({
       variables: {
         shippingCharge: parseFloat(shippingRate.input[0].absoluteMoneyCost),
         freeDeliveryThreshold: freeDeliveryThreshold !== 0 && isActive ? freeDeliveryThreshold.toString() : ""
       }
+    }).catch(err=>{
+      console.error(err)
     })
 
     await updateShippingPolicy({variables: {
       ...shippingPolicy
     }}).catch(err => {
+      setIsLoading(false);
       console.error(err)
       showToast({
         message: intl.formatMessage({
```

Comment:
I wouldn't mix `async/await` with `then/catch`

A big `try/catch` block would do the trick. You could add the `setIsLoading` call inside the `finally` block

Reply:
Good point, done